### DISCUSSION TITLE
React

### DISCUSSION THREAD

**Discussion by mmichelli** [h2oai/wave#1886] [Q&A] [answered]
I need to add a couple of react components, a video player and a subtitle editor. 

Can I extend the h2o_wave.ui library?

**Comment by mturoci** [comment, marked as answer]
Hi @mmichelli, this is something that is currently not supported natively. However, the integration is possible via [custom JS](https://wave.h2o.ai/docs/javascript/) and `ui.markup_card`. 

A couple of blog posts to give you an idea of how to tackle this:

* https://medium.com/@unusualcode/record-audio-in-h2o-wave-apps-fcd6c0b05e59
* https://medium.com/@unusualcode/integrating-vscode-editor-into-h2o-wave-2c075a303974

**Reply by mmichelli** [reply]
Thanks :)
